src/util/common/hashmap.hpp

```diff
@@ -41,9 +33,7 @@ namespace cbdc::hashing {
     template<>
     struct const_sip_hash<buffer> {
         auto operator()(const buffer& buf) const noexcept -> size_t {
-            // TODO: pick the initialization key at random.
-            static constexpr std::array<uint64_t, 2> siphash_key{0x1337,
-                                                                 0x1337};
+            std::array<uint64_t, 2> siphash_key{randombytes_random(), randombytes_random()};
```

Comment:
> My main question is whether libsodium is the most appropriate choice here, so if you have thoughts, I'd appreciate hearing them.

When in-doubt, libsodium (or [monocypher](https://monocypher.org/), in my opinion) are a very good fallback.

However, we already have our own [`random_source`](https://github.com/mit-dci/opencbdc-tx/blob/trunk/src/util/common/random_source.hpp#L27) which we use throughout the codebase. Since the purpose of choosing this key at-random is to avoid collision (rather than specifically a given security property with specific requirements), it probably makes more sense to just leverage that our utility directly.

> Also curious whether this seems appropriate. Since the random values can't be determined at compile time, this update required making this value non-constant. Dynamic RNG seems like a good choice to me, but curious if static would be more appropriate here.

I suppose my immediate wonder is if it would be possible to inject values into that process during the test harness. That way, we could keep our deterministic tests, but still leverage the randomness at runtime. However, we would have then created a divergence between what we're testing and what we're running—and it might be more prudent to modify the tests to handle the randomness (to remain inline).

Mull on that possibility for a little while, and I will attempt to take a look at the affected tests in reasonably short order to have a better handle on the lift.

Reply:
Ah I completely neglected `random_source` -- jumped the gun on that one. One potential solution would be in testing, to seed `random_source` from a fixed file, rather than /dev/urandom as random_source.hpp suggests. However, this does still create divergence. Another, likely more difficult strategy would be when a random value is generated in a test, save it (either in a temporary variable or in a file) for evaluating whether the operation was performed correctly after the fact. 

Also upon further inspection, some of the unit tests that are failing are not ones which I'd expect to fail on first glance, so I will need to delve into that more as well.